### PR TITLE
read and output total aod, dust, SS, SU, OC, and BC aod  to grib2 pass regression test at Hera and DELL

### LINKED ISSUE #258: Reading aerosol aod of total, dust, sea salt, sulfate, organic carbon, and black carbon from GFSV17/GEFSV13 and output to grib2 files
Reading aerosol aod of total, dust, sea salt, sulfate, organic carbon, and black carbon from GFSV16/17 output and output to grib2 files. setting RDAOD to true to turning off calculation of AOD in postprocessing.

### PR DESCRIPTION
**Pull request by AnningCheng-NOAA**
read total aod, dust, SS, SU, OC, and BC aods from netcdf file and output to grib2 files. Regression tests have been passed at DELL: /gpfs/dell2/emc/modeling/noscrub/Anning.Cheng/EMC_post, and at Hera: /scratch2/NCEPDEV/climate/Anning.Cheng/EMC_post. New baseline is needed and was produced at post_regression_test_new/data_out.

### CODE REVIEW

**Review by zhanglikate — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
@zhanglikate You did the changes for outputting aerosol related fields for GEFS V12. Do you think the six new fields Anning has been proposing for MERRA2 aerosol could be unified in block "if (gocart_on)" (starting at line 4433) in subroutine CLDRAD.f? Or making them separated (gocart_on, and RDAOD)? Thanks!

**Comment by zhanglikate** [comment]
Hi  Weng,

         It  can not  use  gocart_on, that different AOD generated  from FV3GFS online calculation based on  MERRA climatological  aerosol data, not the online coupling chemical model. It should only use  RDAOD. Thanks.

Kate

**Comment by WenMeng-NOAA** [comment]
@zhanglikate Thanks for clarifying.

**Comment by WenMeng-NOAA** [comment]
@AnningCheng-NOAA Could you sync your branch with latest changes in develop? The procedure can be found at
https://github.com/NOAA-EMC/EMC_post/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-original-repository

File: sorc/ncep_post.fd/WRFPOST.f [hunk 1]
```diff
@@ -769,11 +770,9 @@ PROGRAM WRFPOST
             CALL INITPOST_NMM
           ELSE IF (MODELNAME == 'FV3R') THEN
 ! use netcdf library to read output directly
-            spval = 9.99e20
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/WRFPOST.f]: Please do not delete this line.

File: sorc/ncep_post.fd/WRFPOST.f [hunk 2]
```diff
@@ -769,11 +770,9 @@ PROGRAM WRFPOST
...
             print*,'CALLING INITPOST_NETCDF'
             CALL INITPOST_NETCDF(ncid3d)
           ELSE IF (MODELNAME == 'GFS') THEN
-            spval = 9.99e20
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/WRFPOST.f]: Please do not delete this line.

File: sorc/ncep_post.fd/WRFPOST.f [hunk 3]
```diff
@@ -782,7 +781,6 @@ PROGRAM WRFPOST
           END IF
 ! use netcdf_parallel library to read fv3 output
         ELSE IF(TRIM(IOFORM) == 'netcdfpara') THEN
-          spval = 9.99e20
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/WRFPOST.f]: please do not delete this line.

File: sorc/ncep_post.fd/CLDRAD.f [hunk 4]
```diff
@@ -922,11 +926,7 @@ SUBROUTINE CLDRAD
         DO L=LM,1,-1
           DO J=JSTA,JEND
             DO I=1,IM
-              if(CFR(I,J,L)<spval) then
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 5]
```diff
@@ -922,11 +926,7 @@ SUBROUTINE CLDRAD
...
               FULL_CLD(I,J)=CFR(I,J,L)    !- 3D cloud fraction (from radiation)
-              else
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 6]
```diff
@@ -922,11 +926,7 @@ SUBROUTINE CLDRAD
...
-              FULL_CLD(I,J)=spval
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 7]
```diff
@@ -936,18 +936,13 @@ SUBROUTINE CLDRAD
               FRAC=0.
               DO JC=max(1,J-numr),min(JM,J+numr)
                 DO IC=max(1,I-numr),min(IM,I+numr)
-!                  if(IC>=1.and.IC<=IM.and.JM>=JSTA.and.JM<=JEND) then
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 8]
```diff
@@ -936,18 +936,13 @@ SUBROUTINE CLDRAD
...
                   IF(FULL_CLD(IC,JC) /= SPVAL) THEN
                     NUMPTS=NUMPTS+1
                     FRAC=FRAC+FULL_CLD(IC,JC)
                   ENDIF
-!                  else
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 9]
```diff
@@ -957,12 +952,6 @@ SUBROUTINE CLDRAD
                 CFRACH(I,J)=MAX(CFRACH(I,J),FRAC)
               ENDIF
               TCLD(I,J)=MAX(TCLD(I,J),FRAC)
-              else
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 10]
```diff
@@ -2490,22 +2479,9 @@ SUBROUTINE CLDRAD
           DO I=1,IM
             ITOP=ITOPT(I,J)
             IF (ITOP>0 .AND. ITOP<=NINT(LMH(I,J))) THEN
-              IF(T(I,J,ITOP)<SPVAL .AND. &
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: sorc/ncep_post.fd/CLDRAD.f [hunk 11]
```diff
@@ -2490,22 +2479,9 @@ SUBROUTINE CLDRAD
...
-                 PMID(I,J,ITOP)<SPVAL .AND. &
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f]: May I ask the reason of deleting these lines?

File: scripts/exgfs_atmos_nceppost.sh [hunk 12]
```diff
@@ -75,7 +75,7 @@ export machine=${machine:-WCOSS_C}
 ###########################
 # Specify Output layers
 ###########################
-export POSTGPVARS="KPO=57,PO=1000.,975.,950.,925.,900.,875.,850.,825.,800.,775.,750.,725.,700.,675.,650.,625.,600.,575.,550.,525.,500.,475.,450.,425.,400.,375.,350.,325.,300.,275.,250.,225.,200.,175.,150.,125.,100.,70.,50.,40.,30.,20.,15.,10.,7.,5.,3.,2.,1.,0.7,0.4,0.2,0.1,0.07,0.04,0.02,0.01,"
+export POSTGPVARS="KPO=57,PO=1000.,975.,950.,925.,900.,875.,850.,825.,800.,775.,750.,725.,700.,675.,650.,625.,600.,575.,550.,525.,500.,475.,450.,425.,400.,375.,350.,325.,300.,275.,250.,225.,200.,175.,150.,125.,100.,70.,50.,40.,30.,20.,15.,10.,7.,5.,3.,2.,1.,0.7,0.4,0.2,0.1,0.07,0.04,0.02,0.01,rdaod=.false.,"
```
Comment (WenMeng-NOAA) [inline scripts/exgfs_atmos_nceppost.sh]: Please revert the change "rdaod=.false.,"

**Comment by AnningCheng-NOAA** [comment]
rdaod=.false. in my branch. Do you mean to delete it?

**Comment by WenMeng-NOAA** [comment]
@AnningCheng-NOAA Yes, please.

**Comment by AnningCheng-NOAA** [comment]
done

File: scripts/exgfs_atmos_nceppost.sh [hunk 13]
```diff
@@ -75,7 +75,7 @@ export machine=${machine:-WCOSS_C}
 ###########################
 # Specify Output layers
 ###########################
-export POSTGPVARS="KPO=57,PO=1000.,975.,950.,925.,900.,875.,850.,825.,800.,775.,750.,725.,700.,675.,650.,625.,600.,575.,550.,525.,500.,475.,450.,425.,400.,375.,350.,325.,300.,275.,250.,225.,200.,175.,150.,125.,100.,70.,50.,40.,30.,20.,15.,10.,7.,5.,3.,2.,1.,0.7,0.4,0.2,0.1,0.07,0.04,0.02,0.01,"
+export POSTGPVARS="KPO=57,PO=1000.,975.,950.,925.,900.,875.,850.,825.,800.,775.,750.,725.,700.,675.,650.,625.,600.,575.,550.,525.,500.,475.,450.,425.,400.,375.,350.,325.,300.,275.,250.,225.,200.,175.,150.,125.,100.,70.,50.,40.,30.,20.,15.,10.,7.,5.,3.,2.,1.,0.7,0.4,0.2,0.1,0.07,0.04,0.02,0.01"
```
Comment (WenMeng-NOAA) [inline scripts/exgfs_atmos_nceppost.sh]: Please keep the comma, change 0.01 into 0.01,

**Comment by WenMeng-NOAA** [comment]
@AnningCheng-NOAA After you commit exgfs_atmos_nceppost.sh, the last step would be sync your branch with upstream/develop via the procedure at
https://github.com/NOAA-EMC/EMC_post/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-original-repository

**Comment by AnningCheng-NOAA** [comment]
Both have been done. Please let me know if you  have any questions

**Comment by WenMeng-NOAA** [comment]
@AnningCheng-NOAA Now your PR is good to me. I will run the final tests. Thanks!

**Comment by WenMeng-NOAA** [comment]
The UPP regression tests were conducted on WCOSS-Dell, WCOSS-Cray and Hera. No changed results were found.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA Please review the changes for exgdas_atmos_nceppost.sh and exgfs_atmos_nceppost.sh.

**Review by KateFriedman-NOAA — APPROVED** [review]
Changes to scripts/exgdas_atmos_nceppost.sh and scripts/exgfs_atmos_nceppost.sh for updating the POSTGRB2TBL setting with updated g2tmpl module paths looks good. Thanks!

**Review by WenMeng-NOAA — APPROVED** [review]
